nbqa/__main__.py

```diff
@@ -451,9 +460,14 @@ def _run_command(
         env=env,
     )
 
-    after = _get_mtimes(arg)
-    mutated = after != before
+    mutated = _get_mtimes(arg) != before
 
+    if command == "black" and mutated and trailing_semicolons:
+        for line_number, line in enumerate(fileinput.FileInput(str(arg), inplace=True)):
+            if line_number in trailing_semicolons:
+                print(f"{line.rstrip()};{os.linesep}", end="")
```

Comment:
Suppose I have a line whose length longer than the `max-line-length` configured for black. In such cases, black breaks a line in to multiple lines.

Before
```Python
print(" ".join(["When", "this", "line", "is", "longer", "than", "the", "black's", "line", "length"]));
```

After formatting

```Python
print(
    " ".join(
        [
            "When",
            "this",
            "line",
            "is",
            "longer",
            "than",
            "the",
            "black's",
            "line",
            "length",
        ]
    )
)
```

This logic would break in such cases. Please confirm with the tests

Reply:
@girip11 yes, you're absolutely right...I'm tempted to say that the solution, then is to only add a semicolon back in if it's at the end of a _cell_, rather than if it's at the end of a line

Reply:
Shall we replace `;` with some magic string like `#SEMI` before formatting(preprocessing) and uncomment/replace #SEMI with `;` after formatting is done(post processing) ?

Reply:
I think we can keep track of which cells originally had trailing semi-colons and then add them back in if necessary, will update the PR now

Reply:
@girip11 Have updated...thank you _so_ much for spotting the issue in my previous attempt at this, that would've been a bad bug :flushed: